ui/org.eclipse.pde.launching/src/org/eclipse/pde/internal/launching/launcher/BundleLauncherHelper.java

```diff
@@ -831,6 +837,117 @@ public static Map<IPluginModelBase, AdditionalPluginData> getAdditionalPlugins(I
 		return resolvedAdditionalPlugins;
 	}
 
+	public static void addRequiredJunitRuntimePlugins(ILaunchConfiguration configuration, Map<String, List<IPluginModelBase>> collectedModels, Map<IPluginModelBase, String> startLevelMap) throws CoreException {
+		// First we collect all required plugins we need to launch the junit
+		// framework...
+		Collection<String> requiredJunitRuntimePlugins = JUnitLaunchConfigurationDelegate.getRequiredJunitRuntimePlugins(configuration);
+		List<IPluginModelBase> collected = new ArrayList<>(requiredJunitRuntimePlugins.size());
+		for (String id : requiredJunitRuntimePlugins) {
+			// we need to add them all to make sure the collection is complete for adding
+			// compatible ones...
+			addIfAbsent(id, collectedModels, startLevelMap).or(() -> collectedModels.getOrDefault(id, List.of()).stream().filter(m -> m.getBundleDescription().isResolved()).findFirst()).ifPresent(collected::add);
+		}
+		// Now collect the dependencies of the collected...
+		Set<BundleDescription> closure = DependencyManager.findRequirementsClosure(collected.stream().map(IPluginModelBase::getBundleDescription).toList());
+		for (BundleDescription description : closure) {
+			collected.add(addIfAbsent(description, collectedModels, startLevelMap));
+		}
+		// Now add any indirectly required items
+		if (collectedModels.containsKey("junit-platform-runner")) { //$NON-NLS-1$
```

Comment:
This entire block is actually not necessary as the dependencies added below are already part of the transitive hull of the o.e.jdt.junitX.runtime bundles. Therefore it's remove din #2013.

Reply:
Is this really always the case? What bundles require this already? Of course if we can omit this case that's even better.